core-plugins/src/main/java/co/cask/hydrator/plugin/batch/source/ExcelInputReader.java

```diff
@@ -312,12 +313,21 @@ private void createDatasets(@Nullable PipelineConfigurer pipelineConfigurer, @Nu
         Map<String, String> properties = new HashMap<>();
         properties.put(Properties.Table.PROPERTY_SCHEMA, errorRecordSchema.toString());
         properties.put(Properties.Table.PROPERTY_SCHEMA_ROW_FIELD, KEY);
-        DatasetProperties datasetProperties = DatasetProperties.builder().addAll(properties).build();
+        ExploreProperties.Builder builder = ExploreProperties.builder();
+        if (!Strings.isNullOrEmpty(excelInputreaderConfig.errorDatasetExplorer)) {
```

Comment:
Value for 'errorDatasetExplorer' is based on value of 'errorDatasetName' and there should be no default value for 'errorDatasetName', that's the reason 'errorDatasetExplorer' could not be set in the default constructor.